### PR TITLE
Cleaner MetadataSyntax tests

### PR DESCRIPTION
**Pull request by peter1000**

### CODE REVIEW

**Comment by peter1000** [comment]
this will fail on current master because it seems to have a [bug](https://github.com/MichaelHatherly/Docile.jl/commit/a1cfb5a50099aa2d1c547a7ddcb959d8424dd838#commitcomment-11132667)

```
Formats.
Docile: updating package list…
Docile: caching 'MetadataSyntax'…
 - MetadataSyntax ✓
     - MetadataSyntax.
  Error   :: (line:-1) :: MetadataSyntax.
    allparsed[fmeth(MetadataSyntax.example1)] => example1: No meta-syntax.
KeyError: example1() not found

```

But it finishes on the previous I tested.

**Comment by MichaelHatherly** [comment]
I'm just working on the AppVeyor failure, path sep problem.

File: test/Formats/facts.jl [hunk 1]
```diff
@@ -1,75 +1,26 @@
...
+require(joinpath(dirname(@__FILE__), "MetadataSyntax.jl"))
 
-You can use metadata in the documentation by including the `Docile metadata syntax` like this:
+using MetadataSyntax, Docile
 
-Example:
+facts("Formats.") do
 
-    \\!!section(Lexicon.md/Exported/Methods/1) this will be used to extract the final Section for the defined docstring.
+  allparsed = Docile.Cache.getparsed(MetadataSyntax)
```
Comment (MichaelHatherly) [inline test/Formats/facts.jl]: 4 space indent.

File: test/Formats/MetadataSyntax.jl [hunk 2]
```diff
@@ -75,10 +69,15 @@ Example:
 
 !!section(Lexicon.md/Exported/Methods/1)
 """
-function example14() end
+function example12() end
```
Comment (MichaelHatherly) [inline test/Formats/MetadataSyntax.jl]: As I mentioned in the other comments on that recent commit, changing these definitions to have something in the body, or just using single line ones will get them picked up.

**Comment by MichaelHatherly** [comment]
The windows trouble is fixed by efad5842cdf72576a31c8bf5b44cb140254eff2c.

**Comment by peter1000** [comment]
I will correct this and update the PR - adjust them a bit to fit better together with the other tests you added.

**Comment by MichaelHatherly** [comment]
Great, thanks. Be aware that travis is being a bit slow currently, still waiting on the last build to finish.

**Comment by peter1000** [comment]
updated PR

File: test/Formats/facts.jl [hunk 3]
```diff
@@ -1,113 +1,58 @@
...
-Pages or [Read the Docs](https://readthedocs.org/).
-
-You can use metadata in the documentation by including the `Docile metadata syntax` like this:
-
-Example:
+    context("No meta-syntax.") do
 
-    \\!!section(Lexicon.md/Exported/Methods/1) this will be used to extract the final Section for the defined docstring.
+        @fact Cache.getparsed(MetadataSyntax, :no_meta_syntax)                                => "No meta-syntax."
+        @fact_throws KeyError Cache.getmeta(MetadataSyntax, :no_meta_syntax)[:no_meta_syntax] => "No meta-syntax."
```
Comment (MichaelHatherly) [inline test/Formats/facts.jl]: `@fact_throws` doesn't take a `=> result` part. Just:

```jl
@fact_throws KeyError Cache.getmeta(MetadataSyntax, :no_meta_syntax)[:no_meta_syntax]
```

**Comment by peter1000** [comment]
the last one has anerr I will fix now

File: test/Formats/MetadataSyntax.jl [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+"!!setget (space_between_metaname_bracket:Space between metaname and bracket is not a metasyntax.)"
+space_between_metaname_bracket = ()
+
+"!!setget(license:[MIT](https://github.com/MichaelHatherly/Lexicon.jl/blob/master/LICENSE.md))"
+brackets_within_meta = ()
+
+"!!setget(笔者:所以不多说了)"
+chinese_unicode = ()
+
+"\\!!setget(russian:бежал мета)"
```
Comment (MichaelHatherly) [inline test/Formats/MetadataSyntax.jl]: lol

**Comment by MichaelHatherly** [comment]
Much appreciated! Thanks.

**Pull request merged**